tinker_cookbook/renderers/deepseek_v3.py

```diff
@@ -363,7 +361,73 @@ def create_conversation_prefix_with_tools(
         return [Message(role="system", content=system_prompt + tools_text)]
 
 
-class DeepSeekV3DisableThinkingRenderer(DeepSeekV3ThinkingRenderer):
+class DeepSeekV3ThinkingRenderer(_DeepSeekV3BaseRenderer):
+    """
+    Renderer for DeepSeek V3 models in THINKING mode.
+
+    Format:
+        <|begin_of_sentence|><|User|>question<|Assistant|><think>reasoning</think>answer<|end_of_sentence|>
+
+    For non-thinking mode, use DeepSeekV3DisableThinkingRenderer instead.
+
+    Generation prompts include <think> to trigger thinking mode, matching HF template
+    behavior with thinking=True. Historical assistant messages get </think> to signal
+    thinking was stripped (when strip_thinking_from_history=True).
+    """
+
+    def render_message(self, message: Message, ctx: RenderContext) -> RenderedMessage:
+        """Render message with </think> header for assistant messages.
+
+        When strip_thinking_from_history=True (default), ALL assistant messages
+        (not following tool) get </think> added to their header. This matches
+        HF template behavior with thinking=True.
+        """
+        rendered = super().render_message(message, ctx)
+
+        # Add </think> to header for assistant messages with content (not following tool)
```

Comment:
are you implying that messages that follow the tool don't have content? why not?